sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -624,6 +630,52 @@ class EthService(
     }
   }
 
+  @RpcMethod("debug_traceCall")
+  @RpcOptionalParameters(1)
+  def traceCall(tx: TransactionArgs, tag: String, config: TraceOptions): Any = {
+
+    applyOnAccountView { nodeView =>
+      // retrieve block hash from input
+      val hash = getBlockHashByTag(nodeView, tag)
+
+      // get block to trace
+      val (block, blockInfo) = getBlockById(nodeView, bytesToId(hash.toBytes))
+
+      // get state at previous block
+      getStateViewAtTag(nodeView, (blockInfo.height - 1).toString) { (tagStateView, blockContext) =>
+        // use default trace params if none are given
+        blockContext.setTraceParams(if (config == null) new TraceOptions() else config)
+
+        // apply mainchain references
+        for (mcBlockRefData <- block.mainchainBlockReferencesData) {
+          tagStateView.applyMainchainBlockReferenceData(mcBlockRefData)
+        }
+
+        val gasPool = new GasPool(block.header.gasLimit)
+
+        // apply block transactions without tracing
+        for ((tx, i) <- block.transactions.zipWithIndex) {
+          tagStateView.applyTransaction(tx, i, gasPool, blockContext)
+        }
+
+        // declare the ethereum transaction from the input parameters
+        val ethTx = tx.toTransaction(networkParams)
+        val ethTxScat = ethTx.asInstanceOf[SidechainTypes#SCAT]
+
+        // use default trace params if none are given
+        blockContext.setTraceParams(if (config == null) new TraceOptions() else config)
+
+        // apply requested transaction with tracing enabled
+        tagStateView.applyTransaction(ethTxScat, block.transactions.length, gasPool, blockContext, signatureRequired = false)
```

Comment:
I would like to change this to the same method we use for `eth_call`:
```
val msg = params.toMessage(blockContext.baseFee, settings.globalRpcGasCap)
tagStateView.applyMessage(msg, new GasPool(msg.getGasLimit), blockContext)
```
Let's not make it look like we have a transaction here, because we don't.